### ISSUE TITLE
[EVOL] Webpack focus for TS

### ISSUE THREAD

**Issue by Tsj-** [KleeGroup/webpack-focus#22]
It would be great to have webpack-focus working for type script project (a.k.a. focus4 project).

Here's my configuration for now:

```
const path = require('path');
const webpack = require('webpack');
const getLocalIdent = require("css-loader/lib/getLocalIdent");
const variables = require("../../css-variables");
const ExtractTextPlugin = require("extract-text-webpack-plugin");

const extractSass = new ExtractTextPlugin({
    filename: "biz.css"
});

module.exports = {
    entry: path.join(__dirname, './common/index'),
    output: {
        path: path.join(__dirname, '../../../api/src/main/webapp/static/biz/'),
        filename: 'biz.js',
        publicPath: ''
    },
    resolve: {
        extensions: ['.js', '.ts', '.tsx']
    },
    module: {
        rules: [
            {
                test: /\.js$/,
                enforce: 'pre',
                use: [{
                    loader: 'source-map-loader'
                }]
            },
            {
                test: /\.tsx?$/,
                include: [__dirname, path.join(__dirname, '../00-core')],
                use: [{
                    loader: 'awesome-typescript-loader'
                }]
            },
            {
                test: /\.json$/,
                use: [{
                    loader: 'json-loader'
                }]
            },
            {
                test: /\.scss$/,
                use: extractSass.extract({
                    fallbackLoader: "style-loader",
                    loader: "css-loader!sass-loader",
                })
            },
            {
                test: /\.css$/,
                use: [
                    { loader: 'style-loader' },
                    {
                        loader: 'css-loader',
                        options: {
                            modules: true,
                            localIdentName: "[path][name]-[local]",
                            getLocalIdent: (loaderContext, localIdentName, localName, options) => {
                                const name = getLocalIdent(loaderContext, localIdentName, localName, options);
                                if (name.includes("focus4")) {
                                    return `focus-${name.split("-").slice(-3).filter(s => s !== "__style__").join("-")}`;
                                } else if (name.includes("toolbox")) {
                                    return `rt-${name.split("-").slice(-3).join("-")}`;
                                } else {
                                    return name;
                                }
                            },
                            importLoaders: 1
                        }
                    },
                    {
                        loader: "postcss-loader",
                        options: {
                            plugins: () => [
                                require("postcss-import")(),
                                require("postcss-cssnext")({
                                    features: {
                                        customProperties: {
                                            variables
                                        }
                                    }
                                })
                            ]
                        }
                    }
                ]
            },
            {
                test: /\.(woff2?|ttf|eot|svg)$/,
                use: [{
                    loader: 'file-loader',
                    options: { name: '[name].[ext]' }
                }]
            },
            {
                test: /\.png$/,
                use: [{
                    loader: 'url-loader',
                    options: { mimetype: 'image/png' }
                }]
            }
        ]
    },
    plugins: [
        extractSass
    ],
    devtool: 'cheap-module-source-map'
};

```
And the package.json:

```
{
  "name": "Spocc",
  "version": "0.0.1",
  "main": "src/index.ts",
  "scripts": {
    "build": "npm run build-biz && npm run build-admin",
    "build-biz": "webpack -p --config app/01-biz/webpack.build.config.js",
    "build-admin": "webpack -p --config app/02-admin/webpack.build.config.js",
    "build-dev": "webpack",
    "clean": "node util/clean.js",
    "stylelint": "stylelint app/**/*.scss app/**/*.css",
    "start": "node dev-server.js",
    "start-biz": "node app/01-biz/dev-server.js",
    "start-admin": "node app/02-admin/dev-server.js",
    "tcm": "tcm src",
    "tslint": "tslint src/**/*.ts src/**/*.tsx --force"
  },
  "dependencies": {
    "flat": "^2.0.1",
    "focus4": "7.0.0"
  },
  "devDependencies": {
    "awesome-typescript-loader": "3.1.3",
    "core-js": "2.4.1",
    "css-loader": "0.28.4",
    "extract-text-webpack-plugin": "2.1.2",
    "file-loader": "0.11.2",
    "json-loader": "0.5.4",
    "node-sass": "4.5.3",
    "react-tooltip": "3.3.0",
    "react-youtube": "7.4.0",
    "sass-loader": "6.0.6",
    "source-map-loader": "0.2.1",
    "style-loader": "0.18.2",
    "stylelint": "7.10.1",
    "stylelint-config-standard": "16.0.0",
    "tslint": "5.4.3",
    "typed-css-modules": "0.2.0",
    "typescript": "2.3.4",
    "url-loader": "0.5.9",
    "webpack": "3.0.0",
    "webpack-dev-server": "2.5.0",
    "react-select": "^1.0.0-rc.3",
    "react-input-autosize": "1.1.0",
    "@types/flat": "0.0.28",
    "@types/react": "15.0.1",
    "@types/react-select": "1.0.46",
    "postcss": "6.0.1",
    "postcss-cssnext": "2.11.0",
    "postcss-import": "10.0.0",
    "postcss-loader": "2.0.5"
  },
  "author": "Kleegroup",
  "license": "MIT"
}

```

**Comment by Hartorn** [comment]
@Tsj- Can this be closed with 1.0.0-beta6 ?

**Comment by Hartorn** [comment]
@Tsj- I'm gonna delete feature-typescript-build soon also.
